packages/core/clients/fallbackKernelAccountClient.ts

```diff
@@ -11,50 +11,79 @@ export const createFallbackKernelAccountClient = <
+                        await onError(error as Error, clients[i].transport.url)
+                    }
+                    if (i === clients.length - 1) {
+                        throw error
+                    }
+                }
+            }
+        }
+    }
+
     const proxyClient = new Proxy(clients[0], {
         get(_target, prop, receiver) {
-            for (const client of clients) {
-                const value = Reflect.get(client, prop, receiver)
-                if (value !== undefined) {
-                    // If the property is a function, wrap it to add fallback logic
-                    if (typeof value === "function") {
-                        // biome-ignore lint/suspicious/noExplicitAny: <explanation>
-                        return async (...args: any[]) => {
-                            for (let i = 0; i < clients.length; i++) {
-                                try {
-                                    const method = Reflect.get(
-                                        clients[i],
-                                        prop,
-                                        receiver
-                                    )
-                                    if (typeof method === "function") {
-                                        // Attempt to call the function on the current client
-                                        return await method(...args)
-                                    }
-                                } catch (error) {
-                                    console.error(
-                                        `Action ${String(
-                                            prop
-                                        )} failed with client ${
-                                            client.transport.url
-                                        }, trying next if available.`,
-                                        error
-                                    )
-                                    if (i === clients.length - 1) {
-                                        throw error
-                                    }
-                                }
+            if (prop === "extend") {
+                // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+                return (fn: any) => {
+                    const modifications = fn(proxyClient)
+                    for (const [key, modification] of Object.entries(
+                        modifications
+                    )) {
+                        if (typeof modification === "function") {
+                            // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+                            ;(proxyClient as any)[key] = async (
+                                // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+                                ...args: any[]
+                            ) => {
+                                return modification(...args)
```

Comment:
I think this should fix the extend issue.  There was an `await` here which was causing the extend function to wrap the return value in a promise

Reply:
oh that makes sense. that's why it does not work after extending. great catch!